### PR TITLE
Fix local links in dev-docs 

### PR DESCRIPTION
**Pull request by soininen**
**Description of work.**

Some of the pages in dev-docs were unaccessible since the `.rst` files had hard-links to the final `.html` pages. Some of this originated because the pages were created in different branches and cross-referencing would have produced warnings with the `dev-docs` target. When the pages were reorganized at some point, the hard-coded links were broken.

This PR changes the hard-coded local links within `dev-docs` to use proper sphinx cross-referencing. Also, some small 'typos' have been corrected here and there.

**Report to:** [nobody].

**To test:**

Build the *dev-docs* target and try to find broken links/broken behavior.

*There is no associated issue.*

*This does not require release notes* because **it is an internal developer documentation change**.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: dev-docs/source/MVPTutorial/AddButton.rst [hunk 1]
```diff
@@ -88,9 +90,9 @@ all been saved in ``view.py``, the ``main.py`` will contain:
     def qapp():
         if QtGui.QApplication.instance():
             _app = QtGui.QApplication.instance()
-	else:
```
Comment (DanNixon) [inline dev-docs/source/MVPTutorial/AddButton.rst]: This has broken the indentation for this example.

File: dev-docs/source/MVPTutorial/PresenterExerciseSolution.rst [hunk 2]
```diff
@@ -52,22 +54,22 @@ View
...
 
-	def buttonPressed(self):
+        def buttonPressed(self):
             self.plotSignal.emit()
 
-	def setTableRow(self, name, row):
+        def setTableRow(self, name, row):
             text = QtGui.QTableWidgetItem(name)
             text.setFlags(QtCore.Qt.ItemIsEnabled)
             col = 0
```
Comment (DanNixon) [inline dev-docs/source/MVPTutorial/PresenterExerciseSolution.rst]: There are two more methods in this class that are not indented correctly.

File: dev-docs/source/MVPTutorial/PresenterExerciseSolution.rst [hunk 3]
```diff
@@ -136,12 +138,12 @@ Main module
...
             self.setWindowTitle("view tutorial")
 
-	def qapp():
+    def qapp():
             if QtGui.QApplication.instance():
                 _app = QtGui.QApplication.instance()
-	    else:
-		_app = QtGui.QApplication(sys.argv)
-	    return _app
+        else:
```
Comment (DanNixon) [inline dev-docs/source/MVPTutorial/PresenterExerciseSolution.rst]: Broken indentation.

File: dev-docs/source/Standards/AlgorithmDocumentation.rst [hunk 4]
```diff
@@ -120,13 +120,16 @@ As the **Description** and **Usage** of an algorithm *cannot* be obtained automa
       return "Rebins data with new X bin boundaries. For EventWorkspaces, you can very quickly rebin in-place by keeping the same output name and PreserveEvents=true.";
    }
 
-``.. alias::``
```
Comment (DanNixon) [inline dev-docs/source/Standards/AlgorithmDocumentation.rst]: Are aliases no longer used?

**Review by DanNixon — CHANGES_REQUESTED** [review]
There are a few Python examples with broken indentation.

File: dev-docs/source/Standards/AlgorithmDocumentation.rst (hunk 4 above)
Comment (soininen) [reply dev-docs/source/Standards/AlgorithmDocumentation.rst]: They're gone.

Comment (DanNixon) [reply dev-docs/source/Standards/AlgorithmDocumentation.rst]: Seems there is at least one algorithm that still uses it (and the ability to define one still exists).

https://github.com/mantidproject/mantid/blob/master/Framework/Algorithms/inc/MantidAlgorithms/ChangeBinOffset.h#L57-L58

Comment (soininen) [reply dev-docs/source/Standards/AlgorithmDocumentation.rst]: Interesting. I didn't find a single `.rst` file with the directive `.. alias::` in it.

Comment (DanNixon) [reply dev-docs/source/Standards/AlgorithmDocumentation.rst]: Ah, it seems aliases are handled by the `relatedalgorithms` directive now. This looks fine in that case.

**Comment by soininen** [comment]
@DanNixon I think I got the issues fixed. Perhaps you could take another look?

**Review by DanNixon — APPROVED** [review]

**Pull request merged**